lib/solid_queue/processes/interruptible.rb

```diff
@@ -7,31 +7,25 @@ def wake_up
     end
 
     private
-      SELF_PIPE_BLOCK_SIZE = 11
 
       def interrupt
-        self_pipe[:writer].write_nonblock(".")
-      rescue Errno::EAGAIN, Errno::EINTR
-        # Ignore writes that would block and retry
-        # if another signal arrived while writing
-        retry
+        queue << true
       end
 
       def interruptible_sleep(time)
-        if time > 0 && self_pipe[:reader].wait_readable(time)
-          loop { self_pipe[:reader].read_nonblock(SELF_PIPE_BLOCK_SIZE) }
-        end
-      rescue Errno::EAGAIN, Errno::EINTR
+        # Since this is invoked on the main thread, using some form of Async
+        # avoids a 35% slowdown (at least when running the test suite).
+        #
+        # Using Futures for architectural consistency with all the other Async in SolidQueue.
```

Comment:
I think you can remove this last comment as it'd make sense more when evaluating the change, but not after the PR has been merged: 
```suggestion
        # Since this is invoked on the main thread, using some form of Async
        # avoids a 35% slowdown (at least when running the test suite).
```

Reply:
@rosa 

I know the SQ team isn't in the code comment camp.  Most, if not all of my comments, were there to help the PR review.  I'm happy to update or remove as preferred.

Reply:
Thank you so much! The comments to help with the PR review are super appreciated! If you want, next time you can also leave the comments using GitHub comments after you've submitted the PR. This is something I do myself, so I don't have to commit the comments to the code but still help with the PR review.